### PR TITLE
Add stateful component HOC 

### PR DESCRIPTION
**Pull request by yordis**
```es6
import {
  createReactNavigationReduxMiddleware,
  createStatefulNavigator
} from "react-navigation-redux-helpers";

const navigationSubscriberKey = "root";
const navStateSelector = state => state.navigation

// 1. create the middleware first
const ReactNavigationMiddleware = createReactNavigationReduxMiddleware(
    navigationSubscriberKey,
    navStateSelector
  );

// 2. add the middleware to your store

// 3. create the HOC

const Navigator = createStatefulNavigator(
    navigationSubscriberKey,
    navStateSelector,
    RootNavigator
  );

// 4. use <Navigator /> from now on
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,12 +13,19 @@
     "invariant": "^2.2.2"
   },
   "devDependencies": {
-    "flow-bin": "^0.64.0"
+    "flow-bin": "^0.64.0",
+    "react-native": "^0.54.0",
```
Comment (yordis) [inline package.json]: I need those so the typespec stop failing

**Comment by yordis** [comment]
@Ashoat could you take this one? It would help to remove boilerplate and it will handle android back button by default.

**Comment by Ashoat** [comment]
@yordis, you're right that this PR would reduce boilerplate. However, I'd actually rather that package users write more boilerplate and explicitly handle things like the back button. It's only a couple lines for them, and having to write those lines reinforces that `react-navigation` does not handle the back button when using Redux integration, and it becomes the package user's job to handle it.

This PR also increases the scope of this package, and consequently the maintenance cost. If RN's back handling changes (like it did recently with `BackAndroid` -> `BackHandler`) I will need to update the code. People in the future might want more advanced back handling, and will submit PRs that I would have to review.

I want to keep the scope of this package limited, and only handle the very basics of Redux integration. Sorry if I'm being a downer - I do appreciate your interest in improving this package. I think this PR would be great in a separate package!

**Comment by yordis** [comment]
@Ashoat so you introduced the HOC at the end of the day.

One of the reason that I didn't want to just use my package is that from consumer of the package I have another breaking changes, while if I was using the HOC component from here probably the integration would be easier.

Anyway, I am glad you did it at the end of the day.

**Pull request closed without merging**